tests/profiler/bob_user_workflow_verbose_profiler_config.yml

```diff
@@ -0,0 +1,216 @@
+variables:
+  max_amount: 99999.99
+  false_positive_threshold: 0.01
+# TODO: Support lists and mappings in variables
+  quantiles:
+    - 0.05
+    - 0.25
+    - 0.50
+    - 0.75
+    - 0.95
+rules:
+  # Here we define 3 rules showing various methods to handle multi-batch parameters
+  # 1. my_rule_for_ids uses a simple ParameterBuilder to build a list of parameters per batch to be aggregated in the referencing ExpectationConfigurationBuilder
+  # 2. my_rule_for_amounts uses an aggregating ParameterBuilder to build a list of parameters and then aggregate them using a specified aggregation function. (While this example is not sufficiently complex to warrant a custom ParameterBuilder, one could be created - alternatively the complexity could be handled in a custom ExpectationConfigurationBuilder)
+  # 3. my_rule_for_numeric_data uses MultiBatchBootstrappedMetricDistributionParameterBuilder to compute multiple parameters to supply ExpectationConfigurationBuilders with pre-computed parameters
+
+  my_rule_for_ids:
+    # Create a value_set based on prior batches (relative to when the profiler is run) and evaluate new batches using this value_set
+    # Rationale: I want alerts when new value_set items are added to my dataset
+    class_name: Rule
+    module_name: great_expectations.profiler.rule.rule
+    domain_builder:
+      # Build list of domains from some categorical column domains: vendor_id, rate_code_id, pickup_location_id, dropoff_location_id, payment_type
+      class_name: SimpleColumnSuffixDomainBuilder
+      module_name: great_expectations.profiler.domain_builder.simple_column_suffix_domain_builder
+      column_name_suffixes:
+        - _id
+        - _type
+    parameter_builders:
+      - name: my_value_set_for_ids
+        # This parameter is the union of all observed batch value_sets
+        # MultiBatchParameterBuilder does not aggregate Parameters, it returns a list of value_set for each batch
+        class_name: MultiBatchParameterBuilder
+        module_name: great_expectations.profiler.parameter_builder.multi_batch_parameter_builder
+        metric_domain_kwargs: $domain.domain_kwargs
+        # Specify all batches prior to the 6 most recent to be used to create this set of Parameters using the below batch_request
+        batch_request:
+          partition_request:
+            partition_index: ":-6"
+        # TODO: Should we also support "offset" from the active_batch (active_batch can be set via passing a validator to the Profiler.profile() method and previously setting the active batch validator.active_batch_id(batch_id="mybatchid"))?
+        # Note: To specify all batches explicitly use the below batch_request
+        # batch_request:
+        #  partition_request:
+        #    partition_index: "0:"
+        # TODO: Maybe this can be omitted and inferred as the default (all batches)? James: maybe default isn't necessarily what's in the validator.
```

Comment:
To which batches do we have access here? Is it only batches we pass into profiler.profile()? Is it all batches available via the data connector? Do we need to pass the data connector to the profiler?

Reply:
Great question. I have been assuming that since in `Profiler.profile()` we require one of four ways to specify batch(es) to profile that these are the assumed batches, the batches a user specifies for profiling. 

One of these ways (batch_request) allows you to specify the data connector 

(docstring of Profiler.profile():
```
Utilizing one of validator, batch, batches, or batch_request (only one may be provided),
        evaluates Profiler object to evaluate rule set on the given data and returns results of rule
        evaluations as an Expectation Suite
```

Reply:
@anthonyburdi @talagluck I have been working under impression that there can be only one `BatchRequest` supplied to `get_validator()` at a time.  This `BatchRequest`, of course, can lead to multiple batches, including the ones whose schemas are not the same.  This is to say that so far, I have not encountered the need for `List[BatchRequest]`.  Hence, a multi-table set of batches can be retrieved with a single  `BatchRequest`.  In addition, the underlying objects, such as `DataConnector` should never be exposed in the user-facing API.